### PR TITLE
Added block pattern for Podcast Grid

### LINKED ISSUE #84: Block Pattern: Post Grid Block
**Is your enhancement related to a problem? Please describe.**
With the advent of [Block Patterns](https://wptavern.com/block-patterns-will-change-everything), we might want to consider a simple block pattern to cover the Post Grid Block concept.

**Describe the solution you'd like**

- [ ] Output grid of episode posts with simple options for how many to render (2,4,6)

**Designs**

TBD, but ideal to help shape what we'd craft

**Describe alternatives you've considered**

n/a

**Additional context**

[Block Pattern creation walkthroughs](https://melchoyce.design/2020/03/30/creating-a-simple-block-pattern-plugin-for-the-gutenberg-editor/) from the super rad @melchoyce.

### PR DESCRIPTION
**Pull request by mehul0810**
### Description of the Change
Added a block pattern for Podcast Grid with the help of Query Block. The figma design attached has Episode and Listening time for which we need to develop those custom blocks to support it.

Closes #84 

### Alternate Designs
Screenshot 2023-01-21 at 8 35 43 AM

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Credits

Props @

### CODE REVIEW

File: includes/block-patterns.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Register block and its assets.
+ */
+function init() {
+	register_block_pattern(
+		'podcasting/podcast-grid',
+		array(
+			'title'       => __( 'Podcast Grid', 'simple-podcasting' ),
+			'description' => _x( 'Podcast Grid', 'This block pattern is used to display podcast in a grid structure.', 'simple-podcasting' ),
+			'content'     => '<!-- wp:query {"queryId":14,"query":{"perPage":3,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"exclude","inherit":false},"displayLayout":{"type":"flex","columns":3}} -->
```
Comment (cadic) [inline includes/block-patterns.php]: The `queryId` property probably needs to be removed here, it will be auto-assigned in the block editor once added.

**Review by cadic — CHANGES_REQUESTED** [review]
@mehul0810 thank you for adding this block pattern.

The code looks good, but still require a few improvements.

Right now the new pattern will contain all posts by default. We could potentially filter the Query Loop block by podcasting taxonomy to only include posts which have a podcast. I suppose adding all podcasts by default will work, `"taxQuery":{"podcasting_podcasts":[42,43,44]}`. The user could reduce the list of podcasts if they want.

Episode and Listening Time blocks expected to be added within the current PR, as well as the cover image for each podcast.

File: includes/block-patterns.php (hunk 1 above)
Comment (noraconley) [reply includes/block-patterns.php]: I made some changes:
- Use podcast terms for query
- Remove query ID
- Adds podcast name
- Use cover block featured image
- Change category for the pattern to query (included in core)

Without building a custom block for it, listening time is not possible with existing core blocks.

**Comment by jeffpaul** [comment]
@nateconley is there more work here or can this proceed for review?

**Comment by noraconley** [comment]
@jeffpaul This can proceed for review if the listening time is okay to omit.

**Comment by jeffpaul** [comment]
@vikrampm1 mind opening a follow-on issue for someone to look into adding the listening time bit?

**Review by dkotter — APPROVED** [review]

**Pull request merged**